### PR TITLE
[NO-JIRA] Add method applyTo to BpkText

### PR DESCRIPTION
**Pull request by tiagoengel**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/master/CONTRIBUTING.md)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

**Comment by georgegillams** [comment]
Thank you for your contribution.
You can use the following dependency to test your application while we review your pull request:

`com.github.Skyscanner:backpack-android:9dab880e34c79f2dc1d4ec98fd3d1689f7c0a9a8`

File: Backpack/src/androidTest/java/net/skyscanner/backpack/text/BpkTextTest.kt [hunk 1]
```diff
@@ -186,4 +188,56 @@ class BpkTextTest {
...
+    font.applyTo(subject)
+
+    Assert.assertEquals(font.typeface, subject.typeface)
+    Assert.assertEquals(font.letterSpacing, subject.letterSpacing)
+    Assert.assertEquals(font.fontSize.toFloat(), subject.textSize)
+  }
+
+  @Test
+  fun applyTo_TextView_withCustomFont() {
+    val withCustomFont = ContextThemeWrapper(context, net.skyscanner.backpack.test.R.style.TestTextCustomFont)
```
Comment (bvitaliyg) [inline Backpack/src/androidTest/java/net/skyscanner/backpack/text/BpkTextTest.kt]: import R.java

File: Backpack/src/androidTest/java/net/skyscanner/backpack/text/BpkTextTest.kt [hunk 2]
```diff
@@ -186,4 +188,56 @@ class BpkTextTest {
...
+    font.applyTo(subject)
+
+    Assert.assertEquals(font.typeface, subject.typeface)
+    Assert.assertEquals(font.letterSpacing, subject.letterSpacing)
+    Assert.assertEquals(font.fontSize.toFloat(), subject.textSize)
+  }
+
+  @Test
+  fun applyTo_Paint_withCustomFont() {
+    val withCustomFont = ContextThemeWrapper(context, net.skyscanner.backpack.test.R.style.TestTextCustomFont)
```
Comment (bvitaliyg) [inline Backpack/src/androidTest/java/net/skyscanner/backpack/text/BpkTextTest.kt]: Import this as well

File: Backpack/src/main/java/net/skyscanner/backpack/button/BpkButton.kt [hunk 3]
```diff
@@ -228,8 +227,7 @@ open class BpkButton : AppCompatButton {
     // If a custom font is set we update the typeface to reflect it.
     // We do not support custom letter spacing for custom fonts at the moment
     if (bpkFont.isCustomFont) {
-      typeface = bpkFont.typeface
-      setTextSize(TypedValue.COMPLEX_UNIT_PX, bpkFont.fontSize.toFloat())
+      bpkFont.applyTo(this)
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/button/BpkButton.kt]: Shall we also update letter spacing here? Previously it was not invoked.

File: Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt [hunk 4]
```diff
@@ -183,7 +187,20 @@ open class BpkText @JvmOverloads constructor(
     val fontSize: Int,
     val letterSpacing: Float?,
     val isCustomFont: Boolean = false
-  )
+  ) {
+
+    fun applyTo(text: TextView) {
+      text.typeface = typeface
+      text.setTextSize(TypedValue.COMPLEX_UNIT_PX, fontSize.toFloat())
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt]: Please consider converting FontDefinition.fontSize type float in the future .

File: Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt [hunk 5]
```diff
@@ -183,7 +187,20 @@ open class BpkText @JvmOverloads constructor(
...
+      letterSpacing?.let { text.letterSpacing = it }
+    }
+
+    fun applyTo(paint: Paint) {
+      paint.typeface = typeface
+      paint.textSize = fontSize.toFloat()
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt]: Math.round?

File: Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt (hunk 4 above)
Comment (bvitaliyg) [reply Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt]: Math.round?

**Comment by georgegillams** [comment]
Thank you for your contribution.
You can use the following dependency to test your application while we review your pull request:

`com.github.Skyscanner:backpack-android:4512c5c190c66aaf0dbdc2adb0fcb13d3937a5d5`

**Pull request merged**